cartridge/graphql/execute.lua

```diff
@@ -248,26 +248,24 @@ local function getFieldEntry(objectType, object, fields, context)
     argumentMap[argument.name.value] = argument
   end
 
-  local defaultValues = {}
   if context.operation.variableDefinitions ~= nil then
+    context.variables = context.variables or {}
     for _, value in ipairs(context.operation.variableDefinitions) do
-      if value.defaultValue ~= nil then
-        local variableType = query_util.typeFromAST(value.type, context.schema)
-        defaultValues[value.variable.name.value] = util.coerceValue(value.defaultValue, variableType)
+      local variable_name = value.variable.name.value
+      if type(context.variables[variable_name]) == 'nil' then
+        local default_value = box.NULL
```

Comment:
looks a bit strange. Are there any cases when we specify incorrect value but get null as default value?

Seems we should raise an error if variable default is not compatible with its value (for some reasons). But not silently change it to null.

Reply:
We raise an error if variable default is not compatible with its value (for some reasons). Now we set default value only if variable value is not present

Reply:
> Now we set default value only if variable value is not present

But it's wrong. It should cause an error. GraphQL has "default variable" feature and we can't simply change absence variable with null. Because in graphql it looks like:
```graphql
query($value: String = null) {  # currently our paser doesn't support null as literal but it's different kind of a problem.
  ...
}
```

Reply:
Now `nil` value - is an error value. If we will use `nil` as default, then if the value of the variable is not specified, then there will be an error

Reply:
`Now nil value - is an error value`

Was it before?
I suggest to preserve old behaviour where it's possible.
So, in case of error you can throw an error.

Reply:
It was the same in parseLiteral

Reply:
Hm... Well, maybe we should simply remove "parseValue" function from places where it was before - to eliminate usage of then and preserve old behaviour.

Your patch should change visible behaviour for persons who implements some custom types.
But it seems not so because of - https://github.com/tarantool/cartridge/pull/1199#issuecomment-749730247 it means you start to pass box.NULL instead of nil value.

Reply:
Okey, I make `nil` as default value of variables